test/init/initialize.test.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable jest/prefer-strict-equal */
+const { promises: fs } = require('fs');
+const initUtils = require('../../dist/src/cmds/init/initutils');
+const readlineUtils = require('../../dist/src/utils/readline');
+const miscUtils = require('../../dist/src/utils/misc');
+const { initHandler } = require('../../dist/src/cmds/init/initialize');
+
+describe('initialize', () => {
+  describe('initHandler', () => {
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+    });
+
+    const mockPackage = {
+      name: 'foo',
+      version: '1.0.0',
+      description: '',
+      main: 'index.js',
+      scripts: { test: 'echo "Error: no test specified" && exit 1' },
+      author: '',
+      license: 'ISC',
+    };
+
+    const mockArgv = { foo: 'bar' };
+    const mockWallet = [
+      {
+        bundle: {},
+        initialPermissions: {},
+      },
+      { dist: 'dist', outfileName: 'bundle.js', port: 8081 },
+    ];
+    const expectedReturnValue = {
+      foo: 'bar',
+      src: 'index.js',
+      dist: 'dist',
+      outfileName: 'bundle.js',
+      port: 8081,
+    };
+
+    it('function successfully executes under normal conditions', async () => {
+      jest.spyOn(console, 'log').mockImplementation();
+      const fsWriteMock = jest.spyOn(fs, 'writeFile').mockImplementation(() => true);
+      const asyncPackageInitMock = jest.spyOn(initUtils, 'asyncPackageInit').mockImplementation(() => mockPackage);
+      const validateEmptyDirMock = jest.spyOn(initUtils, 'validateEmptyDir').mockImplementation();
+      const buildWeb3WalletMock = jest.spyOn(initUtils, 'buildWeb3Wallet').mockImplementation(() => mockWallet);
+      const closePromptMock = jest.spyOn(readlineUtils, 'closePrompt').mockImplementation(() => mockPackage);
+      expect(await initHandler(mockArgv)).toStrictEqual(expectedReturnValue);
+      expect(global.console.log).toHaveBeenCalledTimes(6);
+      expect(fsWriteMock).toHaveBeenCalledTimes(4);
+      expect(asyncPackageInitMock).toHaveBeenCalledTimes(1);
+      expect(validateEmptyDirMock).toHaveBeenCalledTimes(1);
+      expect(closePromptMock).toHaveBeenCalledTimes(1);
+      expect(buildWeb3WalletMock).toHaveBeenCalledWith(mockArgv);
+    });
+
+    it('function logs error when writefile is unsuccessful', async () => {
+      global.snaps = {
+        verboseErrors: false,
+      };
+      jest.spyOn(console, 'log').mockImplementation();
+      const errorMock = jest.spyOn(miscUtils, 'logError').mockImplementation();
+      const fsWriteMock = jest.spyOn(fs, 'writeFile').mockImplementation(() => {
+        throw new Error('error message');
+      });
+      jest.spyOn(process, 'exit').mockImplementation(() => undefined);
```

Comment:
When we expect `process.exit` to be called, we should make it throw an error and expect the call to `initHandler` to reject. In this way, we can avoid checking how many times the various mocks are called, and convince ourselves that errors produce the expected behavior (early termination).